### PR TITLE
OpenRCT2: update to 0.3.2

### PR DESCRIPTION
**Pull request by klardotsh**
Update to 0.3.1, the new title sequences directory, the scripting API, and all the other funsies that came in the new version.

I playtested this for a few minutes online on `x86_64-musl` and I see nothing obviously broken yet.

### CODE REVIEW

File: srcpkgs/OpenRCT2/patches/0001-no-werror.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 2c99d9d41..90b12375a 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -260,7 +260,7 @@ else ()
+ 
+     # Compiler flags
+     set(DEBUG_LEVEL 0 CACHE STRING "Select debug level for compilation. Use value in range 0–3.")
+-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Wall -Wextra -Wshadow")
```
Comment (ericonr) [inline srcpkgs/OpenRCT2/patches/0001-no-werror.patch:9]: You can try to use `vsed` to remove this instead, to avoida patch that may have to be maintained. Also, we should ask upstream to remove it.

File: srcpkgs/OpenRCT2/template [hunk 2]
```diff
@@ -2,33 +2,36 @@
...
+revision=1
 build_style=cmake
-configure_args="$(vopt_if multiplayer "" "-DDISABLE_HTTP_TWITCH=1 -DDISABLE_NETWORK=1")"
+configure_args="-DOPENRCT2_VERSION_TAG=${version} -DDOWNLOAD_TITLE_SEQUENCES=0 -DDOWNLOAD_OBJECTS=0 $(vopt_if multiplayer "" "-DDISABLE_NETWORK=1") $(vopt_if scripting "-DENABLE_SCRIPTING=1" "")"
 hostmakedepends="pkg-config unzip"
 makedepends="SDL2-devel fontconfig-devel freetype-devel libzip-devel
- libpng-devel speexdsp-devel jansson-devel icu-devel zlib-devel
- $(vopt_if multiplayer libcurl-devel) $(vopt_if multiplayer libressl-devel)"
+ libpng-devel speexdsp-devel jansson-devel icu-devel zlib-devel json-c++
+ $(vopt_if multiplayer libcurl-devel) $(vopt_if multiplayer libressl-devel)
```
Comment (ericonr) [inline srcpkgs/OpenRCT2/template:14]: `$(vopt_if multiplayer 'libcurl-devel libressl-devel')`

File: srcpkgs/OpenRCT2/template [hunk 3]
```diff
@@ -2,33 +2,36 @@
...
-_titles_version=0.1.2
-version=0.2.6
-revision=3
+_objects_version=1.0.17
+_titles_version=0.1.2c
+version=0.3.1
+revision=1
 build_style=cmake
-configure_args="$(vopt_if multiplayer "" "-DDISABLE_HTTP_TWITCH=1 -DDISABLE_NETWORK=1")"
+configure_args="-DOPENRCT2_VERSION_TAG=${version} -DDOWNLOAD_TITLE_SEQUENCES=0 -DDOWNLOAD_OBJECTS=0 $(vopt_if multiplayer "" "-DDISABLE_NETWORK=1") $(vopt_if scripting "-DENABLE_SCRIPTING=1" "")"
```
Comment (ericonr) [inline srcpkgs/OpenRCT2/template:10]: break at ~80 columns and use single quotes for the vopts

**Comment by ericonr** [comment]
Cross builds seem to be building fine, I tested armv6l-musl. Didn't test at runtime.

**Comment by klardotsh** [comment]
Since 0.3.2 came out today, I went ahead and bumped this PR to target the new version instead. This required a change to `json-c++` to build the multi-header version rather than the single-header version. Based on [the multi-header version's json.hpp](https://github.com/nlohmann/json/blob/develop/include/nlohmann/json.hpp), this should be a backwards-compatible change.

Game still builds and runs on x86_64-musl.

**Comment by klardotsh** [comment]
I'm not entirely sure what's up with the CI failures. I had that locally, and had to manually build the update to `json-c++`, and I don't know off the top of my head what makes that version dependency inconsistently invalid...

File: srcpkgs/OpenRCT2/template [hunk 4]
```diff
@@ -2,33 +2,42 @@
...
+ -DOPENRCT2_VERSION_TAG=${version}
+ -DDOWNLOAD_TITLE_SEQUENCES=0
+ -DDOWNLOAD_OBJECTS=0
+ $(vopt_if multiplayer '' '-DDISABLE_NETWORK=1')
+ $(vopt_if scripting '-DENABLE_SCRIPTING=1' '')"
 hostmakedepends="pkg-config unzip"
 makedepends="SDL2-devel fontconfig-devel freetype-devel libzip-devel
  libpng-devel speexdsp-devel jansson-devel icu-devel zlib-devel
- $(vopt_if multiplayer libcurl-devel) $(vopt_if multiplayer libressl-devel)"
+ json-c++>=3.9.1_2
```
Comment (ericonr) [inline srcpkgs/OpenRCT2/template:19]: Packages are always built with the versions of dependencies specified in the templates, so you don't need to specify a version here. This is probably what broke the build.

File: srcpkgs/OpenRCT2/template [hunk 5]
```diff
@@ -49,5 +57,7 @@ post_extract() {
 	_srcdir="${XBPS_SRCDISTDIR}/${pkgname}-${version}"
 	mkdir -p data/object data/title
 	unzip -qd data/object "${_srcdir}/objects.zip"
```
Comment (ericonr) [inline srcpkgs/OpenRCT2/template:59]: Please replace the two `unzip` commands with `bsdtar` ones, if possible, then remove `unzip` from `hostmakedepends`.

**Comment by the-maldridge** [comment]
Going to accept this with unzip for now.  It would be nice to use bsdtar, but I ran into a strange bug locally when I tried this.

**Pull request merged**